enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,254 @@
+
+<!-- TODO DEFAULT SNAPSHOT CLASS -->
+
+#### Upgrading
+
+The CSI Snapshot component images are packaged and versioned with each OpenShift release and can be extracted from the ocp-release image. This allows MicroShift to use existing rebase tooling to upgrade the CSI Snapshot components in step with OCP releases.
+
+#### Configuring
+
+Cluster admins will use the VolumeSnapshotClass API to set dynamic configurations for snapshot creation. Driver-specific configuration is available through the ‘parameters’ sub-field, which is a string:string map and is defined by the particular storage provider.
+
+#### Deploying Applications
+
+### API Extensions
+
+CSI Volume Snapshot APIs are a core component of OpenShift Container Platform and are detailed in [OCP documentation](https://docs.openshift.com/container-platform/4.13/storage/container_storage_interface/persistent-storage-csi-snapshots.html).
+
+A portion of the OpenShift documentation is provided below for additional API detail.
+
+#### VolumeSnapshotClass
+
+Allows a cluster administrator to specify different attributes belonging to a VolumeSnapshot object. These attributes may differ among snapshots taken of the same volume on the storage system, in which case they would not be expressed by using the same storage class of a persistent volume claim.
+
+The VolumeSnapshotClass CRD defines the parameters for the csi-external-snapshotter sidecar to use when creating a snapshot. This allows the storage back end to know what kind of snapshot to dynamically create if multiple options are supported.
+
+Dynamically provisioned snapshots use the VolumeSnapshotClass CRD to specify storage-provider-specific parameters to use when creating a snapshot.
+
+The VolumeSnapshotContentClass CRD is not namespaced and is for use by a cluster administrator to enable global configuration options for their storage back end.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: csi-hostpath-snap
+driver: hostpath.csi.k8s.io [1] 
+deletionPolicy: Delete
+```
+
+1. The name of the CSI driver that is used to create snapshots of this **VolumeSnapshotClass** object. The name must be the same as the Provisioner field of the storage class that is responsible for the PVC that is being snapshotted.
+
+#### VolumeSnapshot
+
+Similar to the PersistentVolumeClaim object, the VolumeSnapshot CRD defines a developer request for a snapshot. The CSI Snapshot Controller Operator runs the CSI snapshot controller, which handles the binding of a VolumeSnapshot CRD with an appropriate VolumeSnapshotContent CRD. The binding is a one-to-one mapping.
```

Comment:
Are we bringing the CSI Snapshot Controller Operator into MicroShift?

Reply:
C/p error from OpenShift documentation, will drop.